### PR TITLE
[LOG4J2-2280] Introduce an Apache Pulsar log4j appender

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

Apache Pulsar is a distributed pub/sub messaging system. We are using Pulsar as the message bus to collect and deliver data. Collecting logging data to a message bus like Pulsar is a very typical analytic pipeline. It would be good that log4j provides an appender to append log events into pulsar

*Changes*

This pull request implements an appender using Apache Pulsar.

### CODE REVIEW

File: log4j-pulsar/src/main/java/org/apache/logging/log4j/pulsar/appender/PulsarAppender.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+public final class PulsarAppender extends AbstractAppender {
+
+    /**
+     * Builds PulsarAppender instances.
+     * @param <B> The type to build
+     */
+    public static class Builder<B extends Builder<B>> extends AbstractAppender.Builder<B>
+            implements org.apache.logging.log4j.core.util.Builder<PulsarAppender> {
+
+        @PluginAttribute("key")
```
[the remark this answers is not in the document]
Comment (jvz) [reply log4j-pulsar/src/main/java/org/apache/logging/log4j/pulsar/appender/PulsarAppender.java]: Ok, followup: I see this is how KafkaAppender is written at the moment.

[the remark this answers is not in the document]
Comment (sijie) [reply log4j-pulsar/src/main/java/org/apache/logging/log4j/pulsar/appender/PulsarAppender.java]: @jvz sorry that I am not the expert of log4j. but yeah, I was following what KafkaAppender is doing. Happy to change to the way that is most suitable here if you can point me a few pointers.

**Comment by rgoers** [comment]
My only concern is whether we really want this in the main log4j build. Especially in 3.0 we could break a lot of the "extras" in another repo and build.

**Comment by rgoers** [comment]
Is this really needed given that Apache Pulsar is already distributing a Pulsar Appender? https://github.com/apache/pulsar/tree/master/pulsar-log4j2-appender

**Comment by rgoers** [comment]
Closing this since virtually the same code by the same author is now part of Pulsar itself.

**Pull request closed without merging**